src/Management/src/Prometheus/PrometheusExtensions.cs

```diff
@@ -39,19 +72,80 @@ public static IServiceCollection AddPrometheusActuator(this IServiceCollection s
+    {
+        return UsePrometheusActuator(builder, null);
+    }
+
+    /// <summary>
+    /// Adds the Steeltoe-configured OpenTelemetry Prometheus exporter to an <see cref="IApplicationBuilder" /> instance.
+    /// </summary>
+    /// <param name="builder">
+    /// The <see cref="IApplicationBuilder" />.
+    /// </param>
+    /// <param name="configureBranchedPipeline">
+    /// Optional callback to configure the branched pipeline. Called before registration of the Prometheus middleware.
+    /// </param>
+    /// <returns>
+    /// The incoming <paramref name="builder" /> so that additional calls can be chained.
+    /// </returns>
+    public static IApplicationBuilder UsePrometheusActuator(this IApplicationBuilder builder, Action<IApplicationBuilder>? configureBranchedPipeline)
     {
         ArgumentNullException.ThrowIfNull(builder);
 
-        ManagementOptions? managementOptions = builder.ApplicationServices.GetService<IOptionsMonitor<ManagementOptions>>()?.CurrentValue;
+        ManagementOptions managementOptions = builder.ApplicationServices.GetRequiredService<IOptionsMonitor<ManagementOptions>>().CurrentValue;
+        var conventionOptions = builder.ApplicationServices.GetRequiredService<IOptionsMonitor<ActuatorConventionOptions>>();
 
         PrometheusEndpointOptions? prometheusOptions = builder.ApplicationServices.GetServices<EndpointOptions>()
             .OfType<PrometheusEndpointOptions>().FirstOrDefault();
 
-        string basePath = managementOptions?.Path ?? "/actuator";
+        string basePath = managementOptions.Path ?? "/actuator";
         string endpointPath = prometheusOptions?.Path ?? "prometheus";
         string path = $"{basePath}/{endpointPath}".Replace("//", "/", StringComparison.Ordinal);
 
-        return builder.UseOpenTelemetryPrometheusScrapingEndpoint(path);
+        builder.UseOpenTelemetryPrometheusScrapingEndpoint(null, null, path, ConfigureBranchedPipeline, null);
+
+        if (Platform.IsCloudFoundry)
+        {
+            var permissionsProvider = builder.ApplicationServices.GetService<PermissionsProvider>();
+
+            if (permissionsProvider is null)
+            {
+                var loggerFactory = builder.ApplicationServices.GetRequiredService<ILoggerFactory>();
+                ILogger logger = loggerFactory.CreateLogger(nameof(PrometheusExtensions));
+                logger.LogWarning("The Cloud Foundry Actuator is required in order to run the Prometheus exporter under the Cloud Foundry context.");
+                return builder;
+            }
+
+            string cloudFoundryPath = $"/{ConfigureManagementOptions.DefaultCloudFoundryPath}/{endpointPath}".Replace("//", "/", StringComparison.Ordinal);
+            builder.UseOpenTelemetryPrometheusScrapingEndpoint(null, null, cloudFoundryPath, ConfigureBranchedPipeline, null);
+        }
+
+        return builder;
+
+        void ConfigureBranchedPipeline(IApplicationBuilder branchedApplicationBuilder)
```

Comment:
It's a bit confusing to have both the `ConfigureBranchedPipeline` and the `configureBranchedPipeline` parameter, which are related but not the same.
What looks wrong is that when `configureBranchedPipeline` is null, it _still_ sets up the pipeline. I think the logic in `ConfigureBranchedPipeline` should move into `ConfigurePrometheusActuatorStartupFilter`, to make it possible to a) take all the defaults, or b) configure everything yourself.

Perhaps it's easier to use the same mechanism as `ActuatorConventionOptions` (store the callback(s) in the container), so users have an extra method to configure the pipeline instead of passing the full callback as a parameter. Such a method could be: `ConfigurePrometheusEndpoint` (the equivalent of `ConfigureActuatorEndpoints`)

Reply:
The code could be cleaned up further and new cases aren't covered by tests, but I suppose this is good enough.

Reply:
Thanks, much better!